services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java

```diff
@@ -118,14 +139,20 @@ public GroupResource getGroupById(@PathParam("id") String id) {
     @Produces(MediaType.APPLICATION_JSON)
     public Map<String, Long> getGroupCount(@QueryParam("search") String search,
                                            @QueryParam("top") @DefaultValue("false") boolean onlyTopGroups) {
-        Long results;
+        List<GroupModel> results;
         Map<String, Long> map = new HashMap<>();
+
         if (Objects.nonNull(search)) {
-            results = realm.getGroupsCountByNameContaining(search);
+            results = realm.searchForGroupByName(search, null, null);
```

Comment:
This is potential performance killer, but I don't have better idea how to address this. So I assume we need to live with that. Fortunately there will be thousands of groups at most (not millions) and it is just about admin requests, so hopefully it is ok.

IMO in general, it is probably better to have pagination approach, which doesn't require counts. Something, which always returns the token for next page? Although this approach also has limitations (EG. it is not easily possible to move to the exact page like "Go directly to page 50" etc), and there are probably other limitations. But maybe something to consider in the design of new REST API (CC @pedroigor ). Sorry for hijack the PR :)

Reply:
@mposolda, the guideline is now updated with the pagination practice. In a nutshell, we are keeping the existing behavior when paginating results with small improvements on how we communicate whether or not the client reached the last page. We can discuss more there :)